server/src/test/java/org/elasticsearch/index/fielddata/FilterFieldDataTests.java

```diff
@@ -52,7 +52,7 @@ public void testFilterByFrequency() throws Exception {
 
         {
             indexService.clearCaches(false, true);
-            MappedFieldType ft = new TextFieldMapper.Builder("high_freq", createDefaultIndexAnalyzers()).fielddata(true)
+            MappedFieldType ft = new TextFieldMapper.Builder("high_freq", createDefaultIndexAnalyzers(), false).fielddata(true)
```

Comment:
I think `indexService.getIndexSettings().getMode().isSyntheticSourceEnabled()` can be used here? This is a test, but it is good to recreate the builder as close as it would be in production.

I think in case of the other test classes this applies too.

Reply:
I have fixed this where it  was possible.